### ISSUE TITLE
Training instability

### ISSUE THREAD

**Issue by AI-Guru** [marcoppasini/musika#12]
Hi!

Just a great work! I am very inspired!

I got started with a tiny dataset. The size is 87 minutes. Preprocessing worked like a charm. Training started smoothly. Kudos BTW for the gradio-during-training idea! Somewhere around 15-20 epochs the training lost stability. As shown in this picture:
<img width="278" alt="grafik" src="https://user-images.githubusercontent.com/32195399/197391510-9709d395-42e0-45f4-bec6-be1d638993e2.png">

Any ideas how I could improve this?

Keep up the great work?

Best,
Tristan

**Comment by marcoppasini** [comment]
Hi Tristan, thank you for trying Musika out!

I think you are training Musika from scratch, correct? If so, 87 minutes is quite a small dataset.
You can try finetuning instead, and training should be much more stable.
In any case, the solution to these instabilities is almost always to lower the learning rate a bit, specifying the --lr argument in the command.

Hope this helps, and if you keep encountering instabilities, please reach out again!
Training on such limited datasets is an interesting scenario that I only briefly experimented with and I am very curious about.

**Comment by AI-Guru** [comment]
Hi Marco!

Thanks a lot! I am doing finetuning now!

A question. There is something that I did not get from the README so far. Do I have to resample my WAV files to 22K before preprocessing when I want to fine tune the misc model?

**Comment by marcoppasini** [comment]
No, Musika actually works with 44.1 kHz samples, and it even automatically resamples them before encoding if they are not 44.1 kHz.
Is there a passage in the Readme which you think is a bit confusing on this topic? I will gladly change it!

Thank you and good luck with your experiments!

**Comment by ModeratePrawn** [comment]
@marcoppasini What was the size of the dataset you used to train the base model?

**Comment by marcoppasini** [comment]
The Misc model (the one used for finetuning by default) was trained on the SXSW dataset, which consists of 17000 songs. Hope this helps!

**Comment by ModeratePrawn** [comment]
Cool, thank you. I am training from scratch on about 40 hours of music. Assuming the average song length is 3 minutes, 17k songs is roughly 850 hours of music for the misc model dataset. Sounds like I should consider adding more.

**Comment by marcoppasini** [comment]
Adding more data certainly always helps! How much it helps really depends on the complexity of the music domain of your dataset.

If you find some interesting insights about how Musika scales with training data feel free to share them!

**Comment by 2feet6inches** [comment]
<img width="808" alt="スクリーンショット 2022-10-24 9 38 23" src="https://user-images.githubusercontent.com/113332846/197426988-32f5c14e-f388-4cb7-b423-a38e5e6576a0.png">
hi! I also got unstable at 15-20 epochs.
I am training with Scratch.
I will try fine-tuning.

**Comment by marcoppasini** [comment]
You could also try to restart training from the last healthy checkpoint!

**Comment by AI-Guru** [comment]
I trained using fine-tuning. Training was very stable. Thanks!

**Comment by marcoppasini** [comment]
Nice to hear, thank you for sharing!
